### ISSUE TITLE
LamRA-rerank 训练 部署问题

### ISSUE THREAD

**Issue by Yimi81** [Code-kunkun/LamRA#24]
作者你好，感谢你出色的工作。我想请问一下为什么train_rerank中的训练模型加载使用的是Qwen2_5_VLForConditionalGeneration，仅在merge_lora的时候Qwen2_5_VLRetForConditionalGeneration。那我在训练rerank的时候其使用的还是Qwen2_5_VLForConditionalGeneration自带的loss吗，然后为什么仅在合并权重的时候使用Qwen2_5_VLRetForConditionalGeneration，这导致我用vllm也没法部署。 @Code-kunkun

**Comment by Code-kunkun** [comment]
如果是rerank阶段，模型其实就是Qwen2_5_VLForConditionalGeneration，使用的是自带的loss.
合并权重的时候，我当时应该是为了方便Ret和Rerank的合并，所以写成了Qwen2_5_VLRetForConditionalGeneration

**Comment by Yimi81** [comment]
那rerank阶段的reranking score如何计算的呢，因为这样的话本质上listwise rerank就是使用当前训练的prompt调了一下vl模型得到最优的商品的序号，这个序号是如何转为分数，最后通过S = α × Sret + (1 − α) × Srank得到最终每个商品的分数呢。

**Comment by Yimi81** [comment]
还想请问一下作者有测试过单独listwise的数据训练和Joint Training for Pointwise and Listwise Reranking，推理的时候只用listwise这个rerank效果会更好吗？论文中似乎没有提及。 感谢回复！

**Comment by Code-kunkun** [comment]
> 那rerank阶段的reranking score如何计算的呢，因为这样的话本质上listwise rerank就是使用当前训练的prompt调了一下vl模型得到最优的商品的序号，这个序号是如何转为分数，最后通过S = α × Sret + (1 − α) × Srank得到最终每个商品的分数呢。

这个你可以考虑计算模型输出 1, 2, 3 ... n 的 logits 做个softmax来当Srank的分数

**Comment by Code-kunkun** [comment]
> 还想请问一下作者有测试过单独listwise的数据训练和Joint Training for Pointwise and Listwise Reranking，推理的时候只用listwise这个rerank效果会更好吗？论文中似乎没有提及。 感谢回复！

这个我有点忘了，我好像没有做单独listwise训练的实验😂
